quantfinlib/train_test_split/kfoldpe.py

```diff
@@ -0,0 +1,197 @@
+        if n_folds < 2:
+            raise ValueError("Number of folds must be at least 2.")
+        self.n_folds = n_folds
+        self.n_purge = n_purge
+        self.n_embargo = n_embargo
+        self.look_forward = look_forward
+
+    def get_n_splits(self) -> int:
+        """Return the number of splits.
+
+        Returns
+        -------
+        int
+            Number of splits.
+        """
+        if self.look_forward:
+            return self.n_folds - 1
+        return self.n_folds
+
+    def split(
+        self,
+        X: Union[np.ndarray, pd.DataFrame],
+        y: Optional[Union[np.ndarray, pd.Series]] = None,
+        groups: Optional[Union[np.ndarray, pd.Series]] = None,
+    ) -> Generator[Tuple[np.ndarray, np.ndarray], None, None]:
+        """Generate indices to split data into training and test sets.
+
+        Parameters
+        ----------
+        X : Union[np.ndarray, pd.DataFrame]
+            The data to split.
+        y : Optional[Union[np.ndarray, pd.Series]], optional
+            The target variable, by default None.
+        groups : Optional[Union[np.ndarray, pd.Series]], optional
+            The groups, indicating datetime. by default None.
+
+        Yields
+        ------
+        Generator[Tuple[np.ndarray, np.ndarray], None, None]
+            Train and test indices.
+        """
+        super().split(X, y, groups)
+        _validate_split_settings(
+            n_samples=X.shape[0], n_folds=self.n_folds, n_embargo=self.n_embargo, n_purge=self.n_purge
+        )
+    
+        if groups is None:
+            groups = np.arange(X.shape[0])
+        indices = np.arange(X.shape[0])
+        assert len(groups) == len(indices), "Bummer! Groups and indices must have the same length."
+        test_indices_per_fold = np.array_split(indices, self.n_folds)
+        # fix the bounds of the folds so that there is no overlap between their groups
+        bnd_per_fold = np.array([[test_indices[0], test_indices[-1]] for test_indices in test_indices_per_fold])
+        bnd_per_fold = modify_fold_bnds(bnd_per_fold=bnd_per_fold, groups=groups)
+        test_indices_per_fold = [np.arange(bnd[0], bnd[1] + 1) for bnd in bnd_per_fold]
+        if self.look_forward:
+            test_indices_per_fold = test_indices_per_fold[1:]
+        for test_index in test_indices_per_fold:
+            train_index = get_train_index_per_split(
+                indices=indices, test_index=test_index, groups=groups, look_forward=self.look_forward
+            )
+            train_index = _purge(
+                train_index=train_index, test_index=test_index, indices=indices, groups=groups, n_purge=self.n_purge
+            )
+            train_index = _embargo(
+                train_index=train_index, test_index=test_index, indices=indices, groups=groups, n_embargo=self.n_embargo
+            )
+            yield train_index, test_index
+
+
+def modify_fold_bnds(bnd_per_fold: np.ndarray, groups: np.ndarray) -> np.ndarray:

```

Comment:
maybe `bounds` or `limits`? `bnds` is not unambiguous.

Reply:
Sounds good. fixed this